client/state/invites/test/selectors.js

```diff
@@ -86,6 +95,85 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( '#getSelectedInvite()', () => {
+		test( 'should return invite', () => {
+			const state = {
+				invites: {
+					items: {
+						12345: [
+							{
+								key: '123456asdf789',
+								role: 'follower',
+								isPending: null,
+								inviteDate: '2018-01-28T17:22:16+00:00',
+								acceptedDate: '2018-01-28T17:22:20+00:00',
+								user: {
+									login: 'chicken',
+									email: false,
+									name: 'Pollo',
+									avatar_URL:
+										'https://2.gravatar.com/avatar/eba3ff8480f481053bbd52b2a08c6136?s=96&d=identicon&r=G',
+								},
+								invitedBy: {
+									login: 'cow',
+									name: 'Vaca',
+									avatar_URL:
+										'https://2.gravatar.com/avatar/e2c5df270c7adcd0f6a70fa9cfde7d0f?s=96&d=identicon&r=G',
+								},
+							},
+						],
+					},
+				},
+			};
+			expect( getSelectedInvite( state, 12345, '123456asdf789' ) ).to.eql(
+				state.invites.items[ 12345 ][ 0 ]
+			);
+		} );
+
+		test( 'should return the same object given the same input', () => {
+			const state = {
+				invites: {
+					items: {
+						12345: [
+							{
+								key: '123456asdf789',
+								role: 'follower',
+								isPending: null,
+								inviteDate: '2018-01-28T17:22:16+00:00',
+								acceptedDate: '2018-01-28T17:22:20+00:00',
+								user: {
+									login: 'chicken',
+									email: false,
+									name: 'Pollo',
+									avatar_URL:
+										'https://2.gravatar.com/avatar/eba3ff8480f481053bbd52b2a08c6136?s=96&d=identicon&r=G',
+								},
+								invitedBy: {
+									login: 'cow',
+									name: 'Vaca',
+									avatar_URL:
+										'https://2.gravatar.com/avatar/e2c5df270c7adcd0f6a70fa9cfde7d0f?s=96&d=identicon&r=G',
+								},
+							},
+						],
+					},
+				},
+			};
+			const call1 = getSelectedInvite( state, 12345, '123456asdf789' );
+			const call2 = getSelectedInvite( state, 12345, '123456asdf789' );
+			expect( call1 ).equal( call2 );
```

Comment:
This test is intended to verify that the selector is memoized as expected (see [previous discussion](https://github.com/Automattic/wp-calypso/pull/21909/files/61590a94bc4dc0bf723784b8c43801dfee311eac#diff-91994670f4a5661c3a90f3a6effff9c7)), but it doesn't work in this case.  Even if the cache is not working, and multiple calls to `find` occur, the selector would still return the same object (the same piece of the state tree).

I think a better way to test the caching would be to observe and count calls to lodash `find`.  I'll give this a try.

cc @samouri, interested to hear your thoughts here.

Reply:
@nylen: you are right that the test doesn't actually test caching, it only tests the prevention of needless re-renders (while oftentimes the same thing, it isn't in this case).

Do you think it would be helpful if a `treeSelect`ed selector had a property on it called `cacheHits` or `cacheMisses`? That would be trivial to add and would mean you could write a test like this:

```jsx
test( 'utilizes the selector cache when called twice with the same input', () => {
  const state = ....;
  
  const cacheHits = getSelectedInvite.cacheHits;
  getSelectedInvite( ... );
  getSelectedInvite( ... );
  expect( getSelectedInvite.cacheHits ).eql( cacheHits + 1 )
} )
```

cc @jsnajdr @dmsnell

Reply:
PS. I think both tests can meaningfully coexist

Reply:
if we do that let's wrap it in a static `process.env.NODE_ENV !== 'production'` condition

Reply:
@dmsnell that was my plan! I'll also be placing the current `clearCache` there as well

Reply:
Adding `cacheHits` and `cacheMisses` properties seems fine to me.  One thing I like about the current approach is that it doesn't require much knowledge of the `treeSelect` code (and would catch bugs with its implementation too), but if this is well-tested in the `treeSelect` tests, then that doesn't matter.